### PR TITLE
Replace use of `Any` by an enum

### LINKED ISSUE #17: Remove uses of `Any`
Currently we use `Any` for handling `Read`- and `Write`Queries. This is undesirable and a better solution might be possible.

It should be figured out if there's a cleaner solution without Runtime Reflection and then that solution should be implemented.

### PR DESCRIPTION
**Pull request by pcpthm**
Fixes #17.
I came from hacktoberfest tag and not familiar with the codebase so I kept in mind to make minimal changes.

### CODE REVIEW

File: src/query/mod.rs [hunk 1]
```diff
@@ -50,6 +50,23 @@ impl fmt::Display for Timestamp {
     }
 }
 
+pub enum InfluxDbQueryTypes<'a> {
```
Comment (Empty2k12) [inline src/query/mod.rs]: Please add some rustdoc comments so it's clear what this is used for for users of this library.

**Comment by Drevoed** [comment]
Was this merge and Pull Request really necessary? We don't have to store references to read/write queries and more importantly, influxdb-rust already had `QueryType`. Code using any could be refactored without the addition of redundant enums. Trait InfluxDbQuery describes a method `get_type` which returns either `QueryType::Read/WriteQuery` depending on a struct.

**Comment by Drevoed** [comment]
@Empty2k12 
```rust
    pub async fn query<Q>(&self, q: &Q) -> Result<String, InfluxDbError>
    where
        Q: InfluxDbQuery
    {
        let query = match q.build() {
            Err(err) => {
                let error = InfluxDbError::InvalidQueryError {
                    error: format!("{}", err),
                };
                return Err(error);
            }
            Ok(query) => query,
        };

        let basic_parameters: Vec<(String, String)> = self.into();

        let client = match q.get_type() {
            QueryType::ReadQuery =>
//...
```
this is how it could and should (imho) be done (why add something to the crate when we already had it) and is how I'm implementing an async-await unstable feature right now

**Comment by pcpthm** [comment]
@Drevoed If I can break API, my opinion is:
- `InfluxDbQuery` trait is unfit because it is not semantically open (it has to be either read or write query, no consuming crate should be implement this trait). It is currently also used for something like a something like a module but is is wrong. Remove it.
- New `InfluxDbQueryTypes` can be used only for the purpose of "overloading" emulation. Not necessary TBH.

**Comment by Drevoed** [comment]
@pcpthm I agree and I've come up with a different solution then, seeing as crate is still in 0.0.* and allows breaking changes. Something along the lines of this:
```rust
pub enum InfluxDbQuery {
    Write(InfluxDbWriteQuery),
    Read(InfluxDbReadQuery)
}

impl InfluxDbQuery {
    pub fn build(&self) -> Result<ValidQuery, InfluxDbError> {
        use InfluxDbQuery::*;
        
        match self {
            Write(write_query) => write_query.build(),
            Read(read_query) => read_query.build()
        }
    }
}
```
this follows your idea of removing an InfluxDbQuery

**Comment by Empty2k12** [comment]
I'm happy to discuss a more semantic API for this crate. Usability and code readability for implementers should be highest concern. I had no one to check the semantics of the API I designed, so I'm thankful for all feedback.

**Pull request merged**